src/main/java/com/google/sps/servlets/GetNearbyVendorsServlet.java

```diff
@@ -0,0 +1,145 @@
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.GeoPt;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.CompositeFilter;
+import com.google.appengine.api.datastore.Query.CompositeFilterOperator;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.gson.Gson;
+import com.google.sps.data.HttpServletUtils;
+import com.google.sps.data.SaleCard;
+import com.google.sps.data.Vendor;
+import java.io.IOException;
+import java.time.LocalTime;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that looks for nearby vendors */
+@WebServlet("/get-nearby-vendors")
+public class GetNearbyVendorsServlet extends HttpServlet {
+
+  private final static float MAX_DISTANCE = 20000f; // 20 kilometers
+  private final static float MIN_DISTANCE = 0f;
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {    
+    // -- The way we get these values might change. This is just an idea ---------------
+    String prefixGeoHash = HttpServletUtils.getParameter(request, "prefixGeoHash", "");
+    boolean hasDelivery = Boolean.parseBoolean(HttpServletUtils.getParameter(request, "hasDelivery", "false"));
+    float latitude = 0f;
+    float longitude = 0f;
+    float distance = 0f;
+    GeoPt clientLocation = new GeoPt(0f, 0f);
+    try {
+      latitude = Float.parseFloat(HttpServletUtils.getParameter(request, "lat", "360"));
+      longitude = Float.parseFloat(HttpServletUtils.getParameter(request, "long", "360"));
+      distance = Float.parseFloat(HttpServletUtils.getParameter(request, "distance", "1000"));
+      clientLocation = new GeoPt(latitude, longitude);
+    } catch (NumberFormatException e) {
+      System.out.println("The string is not a parsable float: " + e);
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST);
+      return;
+    } catch (IllegalArgumentException e) {
+      System.out.println("Latitude and/or longitude outside legal range: " + e);
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST);
+      return;
+    }
+    // ---------------------------------------------------------------------------------
+    
+    // Check values exist and are in the range
+    if (prefixGeoHash.isEmpty() || distance > MAX_DISTANCE || distance < MIN_DISTANCE) {
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST);
```

Comment:
Same here, throw an error code or something that will helps us to identify the error.